### PR TITLE
Alphabetize properties in JS/TS docs and fix typos

### PR DESCRIPTION
**Pull request by conorwastakenwastaken**
Don't mind me farming. Stuff not being alphabetized in some areas, but then alphabetized in other areas is annoying to me

 - Alphabetize properties functions in JS & TS docs
 - Minor typo fixes

### CODE REVIEW

File: lib/gateway/Shard.js [hunk 1]
```diff
@@ -1177,10 +1177,10 @@ class Shard extends EventEmitter {
                 * @prop {Object?} oldMember The old member data, or null if the member wasn't cached
                 * @prop {String?} oldMember.avatar The hash of the member's guild avatar, or null if no guild avatar
                 * @prop {Number?} oldMember.communicationDisabledUntil Timestamp of previous timeout expiry. If `null`, the member was not timed out
-                * @prop {Array<String>} oldMember.roles An array of role IDs this member is a part of
                 * @prop {String?} oldMember.nick The server nickname of the member
                 * @prop {Number?} oldMember.premiumSince Timestamp of when the member boosted the guild
                 * @prop {Boolean?} oldMember.pending Whether the member has passed the guild's Membership Screening requirements
```
Comment (bsian03) [inline lib/gateway/Shard.js]:
```suggestion
                * @prop {Boolean?} oldMember.pending Whether the member has passed the guild's Membership Screening requirements
                * @prop {Number?} oldMember.premiumSince Timestamp of when the member boosted the guild
```

File: lib/rest/Endpoints.js [hunk 2]
```diff
@@ -6,7 +6,6 @@ module.exports.BASE_URL = "/api/v" + REST_VERSION;
 module.exports.CDN_URL = "https://cdn.discordapp.com";
 module.exports.CLIENT_URL = "https://discord.com";
 
-module.exports.ORIGINAL_INTERACTION_RESPONSE =            (appID, interactToken) => `/webhooks/${appID}/${interactToken}`;
 module.exports.COMMAND =                              (applicationID, commandID) => `/applications/${applicationID}/commands/${commandID}`;
 module.exports.COMMANDS =                                        (applicationID) => `/applications/${applicationID}/commands`;
 module.exports.COMMAND_PERMISSIONS =         (applicationID, guildID, commandID) => `/applications/${applicationID}/guilds/${guildID}/commands/${commandID}/permissions`;
```
Comment (bsian03) [inline lib/rest/Endpoints.js]:
```suggestion
module.exports.COMMAND_PERMISSIONS =         (applicationID, guildID, commandID) => `/applications/${applicationID}/guilds/${guildID}/commands/${commandID}/permissions`;
module.exports.COMMANDS =                                        (applicationID) => `/applications/${applicationID}/commands`;
```

File: lib/rest/Endpoints.js
Comment (bsian03) [inline lib/rest/Endpoints.js:1]: user connections and user connection platform on L100/101 need to be switched

File: lib/structures/Guild.js [hunk 3]
```diff
@@ -683,8 +683,8 @@ class Guild extends Base {
 
     /**
     * Edit a guild application command
-    * @arg {String} commandID The command ID
-    * @arg {Array<Object>} command An array of command objects
+    * @arg {String} commandID The command id
```
Comment (bsian03) [inline lib/structures/Guild.js]:
```suggestion
    * @arg {String} commandID The command ID
```

File: lib/structures/Message.js [hunk 4]
```diff
@@ -30,15 +30,15 @@ const User = require("./User");
...
 * @prop {String} messageReference.channelID The id of the channel this message was crossposted from
 * @prop {String?} messageReference.guildID The id of the guild this message was crossposted from
+* @prop {String?} messageReference.messageID The id of the original message this message was crossposted from
 * @prop {Object?} interaction An object containing info about the interaction the message is responding to, if applicable
 * @prop {String} interaction.id The id of the interaction
-* @prop {Number} interaction.type The type of interaction
+* @prop {Member?} interaction.member The member who invoked the interaction
 * @prop {String} interaction.name The name of the command
+* @prop {Number} interaction.type The type of interaction
 * @prop {User} interaction.user The user who invoked the interaction
```
Comment (bsian03) [inline lib/structures/Message.js]: Missed all of these :)

File: index.d.ts [hunk 5]
```diff
@@ -46,7 +53,7 @@ declare namespace Eris {
   type AnyGuildChannel = GuildTextableChannel | AnyVoiceChannel | CategoryChannel;
   type AnyThreadChannel = NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel | ThreadChannel;
   type AnyVoiceChannel = TextVoiceChannel | StageChannel;
-  export type ChannelTypeConversion<T extends GuildChannelTypes> = 
+  export type ChannelTypeConversion<T extends GuildChannelTypes> =
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
  type ChannelTypeConversion<T extends GuildChannelTypes> =
```

File: index.d.ts
Comment (bsian03) [inline index.d.ts:1]: Can you go through references and TS errors in this file, you messed up the merge conflict. Anything that throws an error (e.g. Application Command) where the reference has had a recent change should be removed (though please double check before you do)

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: index.d.ts
Comment (conorwastakenwastaken) [reply index.d.ts:1]: ~~The only errors in the TS file are from another PR afaik, like the one shown in my screenshot here. I fixed anything that I broke with my merge commit already (such as all of the interaction classes being duplicate)~~
Nvm I see what you mean now I think

**Comment by conorwastakenwastaken** [comment]
@bsian03 Should be good on the TS typings side of things. Cleared the last errors up now

**Review by bsian03 — APPROVED** [review]

**Pull request merged**